### PR TITLE
set no caching headers for post request

### LINKED ISSUE #746: Fix caching for post requests
Posts should not be cached

### PR DESCRIPTION
**Pull request by alexander-schranz**
__Tasks:__

- [ ] test coverage
- [ ] gather feedback for my changes
- [x] added changelog line

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | none
| Fixed tickets | fixes #746 
| BC Breaks     | no
| Doc           | none

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 CHANGELOG for Sulu
 ==================
+    * BUGFIX      #681 [WebsiteBUndle]  Fixed none caching for post request
```
Comment (dantleech) [inline CHANGELOG.md]: Shoukd be 

````
CHANGE..
=======..

* dev-master
    * BUGFIX ...
````

File: src/Sulu/Bundle/WebsiteBundle/Controller/WebsiteController.php [hunk 2]
```diff
@@ -72,18 +72,22 @@ protected function renderStructure(
...
-                // set the private and shared max age
-                $response->setMaxAge(240);
-                $response->setSharedMaxAge(960);
-
-                // set reverse-proxy TTL (Symfony HttpCache, Varnish, ...)
-                $response->headers->set(
-                    HttpCache::HEADER_REVERSE_PROXY_TTL,
-                    $response->getAge() + intval($structure->getCacheLifeTime())
-                );
+                if ($this->getRequest()->getMethod() != 'GET') {
```
Comment (dantleech) [inline src/Sulu/Bundle/WebsiteBundle/Controller/WebsiteController.php]: Makes sense to me /cc @chirimoya

Comment (dantleech) [inline src/Sulu/Bundle/WebsiteBundle/Controller/WebsiteController.php]: I would use `!==`

**Comment by chirimoya** [comment]
+1

**Comment by alexander-schranz** [comment]
@dantleech  @chirimoya did change it to ->isMethod('GET')

File: src/Sulu/Bundle/WebsiteBundle/Controller/WebsiteController.php [hunk 3]
```diff
@@ -72,18 +72,22 @@ protected function renderStructure(
...
-                $response->setMaxAge(240);
-                $response->setSharedMaxAge(960);
-
-                // set reverse-proxy TTL (Symfony HttpCache, Varnish, ...)
-                $response->headers->set(
-                    HttpCache::HEADER_REVERSE_PROXY_TTL,
-                    $response->getAge() + intval($structure->getCacheLifeTime())
-                );
+                if (!$this->getRequest()->isMethod('GET')) {
+                    $response->setPrivate();
```
Comment (danrot) [inline src/Sulu/Bundle/WebsiteBundle/Controller/WebsiteController.php]: This is not disabling the cache, just enabling the private cache on the client side. You should probably just check for get and execute the current else part then. However, @dantleech has just move this functionality to a different handler in a PR being merged soon, so be careful that this change is not lost at all.

Comment (dantleech) [inline src/Sulu/Bundle/WebsiteBundle/Controller/WebsiteController.php]: Yeah, this PR is irrelevant now I have added this check to the new event subscriber: https://github.com/sulu-cmf/sulu/pull/653/files#diff-a601e69b47c9329ab040d289dfea324fR100

**Comment by dantleech** [comment]
Closing as will be covered by new cache system

File: src/Sulu/Bundle/WebsiteBundle/Controller/WebsiteController.php (hunk 3 above)
Comment (alexander-schranz) [inline src/Sulu/Bundle/WebsiteBundle/Controller/WebsiteController.php]: thats awesome will @dantleech PRs merged soon?

Comment (danrot) [inline src/Sulu/Bundle/WebsiteBundle/Controller/WebsiteController.php]: Hopefully next week

**Pull request closed without merging**